app/models/driver.rb

```diff
@@ -0,0 +1,29 @@
+class Driver < ApplicationRecord
+  has_many :trips
+
+  validates :name, presence: true
+  validates :vin, presence: true, length: { is: 17 }
+
+  def total_earnings
+    every_costs = self.trips.where.not(cost: nil)
+    return nil if every_costs.empty?
+
+    sum = 0
+    every_costs.each do |trip|
+      if trip.cost <= 1.65 * 100
+        sum += trip.cost * 0.8
+      else
+        sum += (trip.cost - 1.65 * 100) * 0.8
+      end
+    end
+    return (sum / 100).round(2)
+  end
```

Comment:
Just a note that you can use enumerables as well.

```suggestion
  sum = every_costs.sum do |trip|
    if trip.cost.nil?
      0
    elsif trip.cost < 1.65 * 100
      trip.cost * 0.8
    else
      (trip.cost - 1.65 * 100) * 0.8 
  end
```